backend/tests/legacy/test_api_snapshot.py

```diff
@@ -105,9 +108,153 @@ def test_allocations(
+            if key == "donor":
+                unique_donors.add(val)
             if key == "amount":
-                assert int(val) == 1000
+                assert int(val) > 0
+            if key == "proposal":
+                unique_proposals.add(val)
+            print("allocation items: ", allocation.items)
+
+    assert len(unique_donors) == 2
+    assert len(unique_proposals) == 3
+
+
+@pytest.mark.api
+def test_withdrawals(
+    client: Client,
+    deployer: UserAccount,
+    ua_alice: UserAccount,
+    ua_bob: UserAccount,
+    ua_carol: UserAccount,
+):
+    res = client.sync_status()
+    assert res["indexedEpoch"] == res["blockchainEpoch"]
+    assert res["indexedEpoch"] > 0
+
+    alice_proposals = get_proposals_addresses(1)[:3]
+    bob_proposals = get_proposals_addresses(1)[:3]
+    carol_proposals = get_proposals_addresses(1)[:3]
+
+    # fund Octant
+    deployer.fund_octant(
+        address=client.config["WITHDRAWALS_TARGET_CONTRACT_ADDRESS"], value=400
+    )
+
+    # lock GLM for one account
+    deployer.transfer(ua_alice, 10000)
+    ua_alice.lock(10000)
+
+    # lock GLM for one account
+    deployer.transfer(ua_bob, 10000)
+    ua_bob.lock(10000)
+
+    # lock GLM for one account
+    deployer.transfer(ua_carol, 10000)
+    ua_carol.lock(10000)
+
+    # forward time to the beginning of the epoch 2
+    move_to_next_epoch(2)
+
+    # fund the vault (amount here is arbitrary)
+
+    vault.fund(deployer._account, 1000 * 10**18)
+
+    # wait for indexer to catch up
+    epoch_no = wait_for_sync(client, 2)
+    print(f"indexed epoch: {epoch_no}")
+
+    # make a snapshot
+    res = client.pending_snapshot()
+    assert res["epoch"] > 0
+
+    # save account budget for assertion
+    res = client.get_rewards_budget(address=ua_alice.address, epoch=1)
+    alice_budget = int(res["budget"])
+
+    # make empty vote to get personal rewards
+    ua_alice.allocate(0, alice_proposals)
+
+    # save account budget for assertion
+    res = client.get_rewards_budget(address=ua_bob.address, epoch=1)
+    bob_budget = int(res["budget"])
+
+    # make empty vote to get personal rewards
+    ua_bob.allocate(0, bob_proposals)
+
+    # save account budget for assertion
+    res = client.get_rewards_budget(address=ua_carol.address, epoch=1)
+    carol_budget = int(res["budget"])
+
+    # make empty vote to get personal rewards
+    ua_carol.allocate(0, carol_proposals)
+
+    # TODO replace with helper to wait until end of voting
+    move_to_next_epoch(3)
+    epoch_no = wait_for_sync(client, 3)
+    print(f"indexed epoch: {epoch_no}")
+
+    # make a finalized snapshot
+    # res = client.pending_snapshot()
+    res = client.finalized_snapshot()
+    assert res["epoch"] == 1
+
+    # first implementation of waiting for scheduler
+    # TODO below
+    # https://viniciuschiele.github.io/flask-apscheduler/rst/api.html
+    # /scheduler/jobs [GET] > returns json with details of all jobs
+    # /scheduler/jobs/<job_id>/run [POST
```

Comment:
This TODO should be dropped. Below worker is waiting for merkle root to be set. It is strictly better than checking the status of the scheduler job.